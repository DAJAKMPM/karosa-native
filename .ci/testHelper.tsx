import React, { JSXElementConstructor } from "react";
import { View } from "react-native";
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from "@testing-library/react-native";
import { Formik, FormikValues } from "formik";

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  _themeProp: unknown = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };

  const renderApi = render(children, options);
  const wrapper = renderApi.queryByTestId(wrapperTestID || "wrapper");
  return { wrapper, ...renderApi };
};

export const renderWithFormikWrapper = <T extends FormikValues>(
  initialValues: T,
  children: React.ReactElement<any, string | JSXElementConstructor<any>>
) => {
  return renderWithWrapper(
    <Formik
      initialValues={initialValues}
      onSubmit={() => console.log("submit")}
      component={() => children}
    />
  );
};
