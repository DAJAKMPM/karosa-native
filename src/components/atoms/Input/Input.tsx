/**
 *
 * Input
 * @format
 *
 */

import React, { FC } from "react";
import { Input as RnInput } from "react-native-elements";

import type { PropsType } from "./types";
import InputStyles from "./styles";

const Input: FC<PropsType> = (props) => {
  const {
    placeholder,
    value,
    disabled,
    inputContainerStyle,
    keyboardType = "default",
    numberOfLines,
    onChange,
    multiline,
    autoCapitalize,
    secureTextEntry,
    inputStyle,
    placeholderColor,
    containerStyle,
    maxLength,
    autoCompleteType,
    autoCorrect,
    onEndEditing,
  } = props;

  return (
    <RnInput
      value={value}
      placeholder={placeholder}
      disabled={disabled}
      multiline={multiline}
      returnKeyType={
        keyboardType === "number-pad" || keyboardType === "default"
          ? "done"
          : "default"
      }
      inputContainerStyle={[InputStyles.inputContainer, inputContainerStyle]}
      inputStyle={inputStyle as never}
      onEndEditing={onEndEditing}
      autoCorrect={autoCorrect}
      placeholderTextColor={placeholderColor}
      keyboardType={keyboardType}
      numberOfLines={numberOfLines}
      onChangeText={onChange}
      autoCompleteType={autoCompleteType}
      autoCapitalize={autoCapitalize}
      maxLength={maxLength}
      secureTextEntry={secureTextEntry}
      containerStyle={[InputStyles.container, containerStyle]}
    />
  );
};

export default Input;
