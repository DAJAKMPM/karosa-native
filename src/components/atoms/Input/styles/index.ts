/**
 *
 * @format
 *
 */

import { StyleSheet } from "react-native";
import { theme, SPACING } from "@app/styles";

const InputStyles = StyleSheet.create({
  container: {
    backgroundColor: theme.colors.white,
    borderRadius: 3,
    padding: SPACING.lg,
    borderWidth: 1,
    borderStyle: "solid",
    borderColor: theme.colors.dark30,
  },
});

export default InputStyles;
