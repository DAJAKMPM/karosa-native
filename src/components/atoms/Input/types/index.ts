/**
 *
 * @format
 *
 */

import {
  KeyboardTypeOptions,
  StyleProp,
  ViewStyle,
  TextStyle,
  NativeSyntheticEvent,
  TextInputEndEditingEventData,
} from "react-native";

type InputOptions = "none" | "sentences" | "words" | "characters";

type AutoCompleteType =
  | "cc-csc"
  | "cc-exp"
  | "cc-exp-month"
  | "cc-exp-year"
  | "cc-number"
  | "email"
  | "name"
  | "password"
  | "postal-code"
  | "street-address"
  | "tel"
  | "username"
  | "off";

export type PropsType = {
  value: string;
  placeholder?: string;
  onChange?(text: string): void;
  onEndEditing?(e: NativeSyntheticEvent<TextInputEndEditingEventData>): void;
  inputStyle?: StyleProp<TextStyle>;
  inputContainerStyle?: StyleProp<ViewStyle>;
  containerStyle?: StyleProp<ViewStyle>;
  placeholderColor?: string;
  disabled?: boolean;
  autoCorrect?: boolean;
  maxLength?: number;
  multiline?: boolean;
  keyboardType?: KeyboardTypeOptions;
  numberOfLines?: number;
  autoCompleteType?: AutoCompleteType;
  autoCapitalize?: InputOptions;
  secureTextEntry?: boolean;
};
