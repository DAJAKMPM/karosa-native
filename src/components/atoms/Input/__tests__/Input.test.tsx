import React from "react";
import { renderWithWrapper } from "@app/ci/testHelper";
import Input from "../Input";

describe("Input Component", () => {
  it("should match snapshot", () => {
    const { toJSON } = renderWithWrapper(<Input value="test" />);

    expect(toJSON()).toMatchSnapshot();
  });
});
