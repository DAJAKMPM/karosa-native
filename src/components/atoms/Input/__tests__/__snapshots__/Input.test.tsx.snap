// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <View
    style={
      Object {
        "paddingHorizontal": 0,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderBottomWidth": 1,
          "borderColor": "#4F4F4F",
          "borderRadius": 3,
          "borderStyle": "solid",
          "borderWidth": 1,
          "flexDirection": "row",
          "padding": 8,
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        editable={true}
        keyboardType="default"
        rejectResponderTermination={true}
        returnKeyType="done"
        style={
          Object {
            "color": "#242424",
            "flex": 1,
            "fontSize": 18,
            "minHeight": 40,
          }
        }
        testID="RNE__Input__text-input"
        underlineColorAndroid="transparent"
        value="test"
      />
    </View>
    <Text
      style={
        Object {
          "color": "#ff190c",
          "display": "none",
          "fontSize": 12,
          "margin": 5,
        }
      }
    />
  </View>
</View>
`;
